clean/tasks/clean_dns.yml

```diff
@@ -90,64 +90,76 @@
           when: (item.1.set.value is defined)  and  ((item.0.name | regex_replace('-(?!.*-).*')) == (item.1.set.record | regex_replace('^(.*?)\\..*$', '\\1')))  and  (item.0.name == item.1.set.value | regex_replace('^(.*?)\\..*$', '\\1'))
       when: cluster_vars.dns_server == "route53"
 
-
     - name: clean/dns/clouddns | Delete DNS entries from clouddns
       block:
-        - name: clean/dns/clouddns | Get GCP Managed Zone
+        - name: clean/dns/clouddns | Get managed zone(s)
           gcp_dns_managed_zone_info:
             auth_kind: serviceaccount
             dns_name: "{{cluster_vars.dns_nameserver_zone}}"
             project: "{{cluster_vars[buildenv].vpc_host_project_id}}"
             service_account_file: "{{gcp_credentials_file}}"
           register: r__gcp_dns_managed_zone_info
 
-        - block:
-            - assert: { that: "r__gcp_dns_managed_zone_info__non_peered | length <= 1", msg: "Multiple non-peered managed zones with the same DNS is not supported ({{r__gcp_dns_managed_zone_info__non_peered | json_query(\"[].name\") }}) - abort" }
+#        - debug: msg={{r__gcp_dns_managed_zone_info}}
 
-            - name: clean/dns/clouddns | Get DNS entries
-              gcp_dns_resource_record_set_info:
-                auth_kind: serviceaccount
-                managed_zone:
-                  name: "{{r__gcp_dns_managed_zone_info__non_peered.0.name}}"
-                  dnsName: "{{r__gcp_dns_managed_zone_info__non_peered.0.dnsName}}"
-                project: "{{cluster_vars[buildenv].vpc_host_project_id}}"
-                service_account_file: "{{gcp_credentials_file}}"
-              register: r__gcp_dns_resource_record_set_info
+        - name: clean/dns/clouddns | Get all non-peered DNS records for managed zones that match cluster_vars.dns_nameserver_zone
+          gcp_dns_resource_record_set_info:
+            auth_kind: serviceaccount
+            managed_zone:
+              name: "{{item.name}}"
+              dnsName: "{{item.dnsName}}"
+            project: "{{cluster_vars[buildenv].vpc_host_project_id}}"
+            service_account_file: "{{gcp_credentials_file}}"
+          register: r__gcp_dns_resource_record_set_info
+          with_items: "{{r__gcp_dns_managed_zone_info.resources | json_query(\"[?dnsName==`\" + cluster_vars.dns_nameserver_zone + \"` && !(peeringConfig)]\") }}"

```

Comment:
Nice way of avoiding any peered zones 👍  wasn't aware of this trick!

Reply:
`json_query` is your friend!